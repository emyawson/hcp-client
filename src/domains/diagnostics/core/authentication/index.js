export {
  validateSession,
  signOut,
  onRedirectToLogin,
} from './authentication.actions';
export {
  getToken,
  getEc6Token,
  getIsValidSession,
  selectHasTemporaryPassword,
  selectHasLoggedOut,
} from './authentication.selectors';
export {
  LOGIN,
  LOGIN_BASE,
  FIRST_TIME,
  FIRST_TIME_BASE,
  VALIDATE_SESSION,
  VALIDATE_SESSION_BASE,
  RESET_PASSWORD,
  RESET_PASSWORD_BASE,
  SIGN_OUT,
  SIGN_OUT_BASE,
  REDIRECT_TO_LOGIN,
} from './authentication.constants';
