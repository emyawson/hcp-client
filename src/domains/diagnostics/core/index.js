export { validateSession } from './authentication';
export { createModal, MODAL_TYPES } from './modal';
export {
  fetchPatientRequest,
  selectPatient,
  selectPatientDevices,
} from './patient';
export {
  clearPatientDateRange,
  fetchPatientDateRangeRequest,
  onPatientDateRangeChange,
  selectPatientFirstMeasurementDate,
  selectPatientLastMeasurementDate,
  selectPatientStartDate,
  selectPatientEndDate,
} from './patient-date-range';
export {
  createRequestActions,
  createRequestActionTypes,
  REQUEST,
  REQUEST_ANIMATION_DELAY,
  selectOngoingRequests,
} from './request';
export {
  getThresholdsRequest,
  getTimeIntervalsRequest,
  getTrafficLightColorFromStatus,
  TRAFFIC_LIGHT_STATES,
} from './strip-delivery';
