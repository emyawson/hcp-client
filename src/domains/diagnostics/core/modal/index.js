export { createModal } from './modal.actions';
export { MODAL_ACTIONS, MODAL_TYPES } from './modal.constants';
