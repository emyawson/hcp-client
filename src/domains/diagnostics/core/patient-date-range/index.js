export {
  fetchPatientDateRangeRequest,
  onPatientDateRangeChange,
} from './patient-date-range.action';
export {
  selectPatientStartDate,
  selectPatientEndDate,
  selectPatientFirstMeasurementDate,
  selectPatientLastMeasurementDate,
} from './patient-date-range.selector';
