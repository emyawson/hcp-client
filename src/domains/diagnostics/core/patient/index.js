export { fetchPatientRequest } from './patient.action';
export { FETCH_PATIENT } from './patient.constant';
export { selectPatient, selectPatientDevices } from './patient.selector';
