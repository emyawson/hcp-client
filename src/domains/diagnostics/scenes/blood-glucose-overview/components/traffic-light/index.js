export { TrafficLight } from './traffic-light.component.js';
