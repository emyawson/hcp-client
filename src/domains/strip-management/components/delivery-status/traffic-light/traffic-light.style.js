import styled from 'styled-components';

import { constructWrapperBackground } from 'src/components/badge/badge.style';
import {
  AnimationHelper,
  transitions,
  transitionSpeed,
  transitionEasing,
  fadeIn,
  getTrafficLightColorFromStatus,
} from 'src/core';
import { convertPxToRem } from 'src/utils';

const borderRatio = 1 / 16.5;

export const TrafficIconSpan = AnimationHelper.extend`
  animation: ${fadeIn} ${transitionSpeed.slow} ${transitionEasing.enter};
  animation-fill-mode: forwards;
  opacity: 0;
  display: flex;
`;

export const Traffic = styled.div`
  align-items: center;
  border-radius: 50%;
  background-clip: padding-box;
  background-color: ${props => getTrafficLightColorFromStatus(props.status)};
  border-color: ${props =>
    constructWrapperBackground(getTrafficLightColorFromStatus(props.status))};
  border-style: solid;
  border-width: ${props =>
    props.border ? convertPxToRem(props.size * borderRatio) : '0'};
  display: flex;
  height: ${props => convertPxToRem(props.size)};
  justify-content: center;
  transition: ${transitions.default};
  width: ${props => convertPxToRem(props.size)};
`;

Traffic.defaultProps = {
  border: false,
  size: 60,
};
