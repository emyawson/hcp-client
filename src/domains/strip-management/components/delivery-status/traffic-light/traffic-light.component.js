import React from 'react';

import { TRAFFIC_LIGHT_STATES, colors } from 'src/core';
import {
  CheckmarkIcon,
  BellIcon,
  XIcon,
  WhiteLineIcon,
} from 'src/assets/icons';
import { hasValue } from 'src/utils';

import { Traffic, TrafficIconSpan } from './traffic-light.style';

const iconSizeRatio = 0.4;

export const stateToIcon = (status, size, hideIcon = false) => {
  const iconHeight = size * iconSizeRatio;
  switch (status) {
    case TRAFFIC_LIGHT_STATES.DELIVER:
      return (
        <TrafficIconSpan>
          <CheckmarkIcon height={iconHeight} />
        </TrafficIconSpan>
      );

    case TRAFFIC_LIGHT_STATES.DELIVER_WITH_ALERT:
      return (
        <TrafficIconSpan>
          <BellIcon height={iconHeight} />
        </TrafficIconSpan>
      );

    case TRAFFIC_LIGHT_STATES.DO_NOT_DELIVER:
      return (
        <TrafficIconSpan>
          <XIcon height={iconHeight} fillColor={colors.white} />
        </TrafficIconSpan>
      );
    default:
      if (hideIcon) return null;
      return (
        <TrafficIconSpan>
          <WhiteLineIcon height={iconHeight} />
        </TrafficIconSpan>
      );
  }
};

export const TrafficLight = ({
  status,
  noIcon = false,
  border = false,
  size = 60,
}) => {
  const trafficLightStatus = hasValue(status)
    ? status
    : TRAFFIC_LIGHT_STATES.DISABLED;
  return (
    <Traffic status={trafficLightStatus} size={size} border={border}>
      {stateToIcon(status, size, noIcon)}
    </Traffic>
  );
};
