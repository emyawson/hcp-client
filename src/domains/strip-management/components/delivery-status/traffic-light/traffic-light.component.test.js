import React from 'react';
import { shallow } from 'enzyme';

import { TRAFFIC_LIGHT_STATES } from 'src/core';
import {
  CheckmarkIcon,
  BellIcon,
  XIcon,
  WhiteLineIcon,
} from 'src/assets/icons';

import { TrafficLight, stateToIcon } from './traffic-light.component';

describe('Traffic light component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <TrafficLight status={TRAFFIC_LIGHT_STATES.DELIVER} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('Sets status to disabled if none provided', () => {
    const wrapper = shallow(<TrafficLight />);
    expect(wrapper.find(WhiteLineIcon)).toHaveLength(1);
  });
  describe('Traffic light component icon', () => {
    it('Should return checkmark icon if status is deliver', () => {
      const wrapper = shallow(
        <div>{stateToIcon(TRAFFIC_LIGHT_STATES.DELIVER, 24, false)}</div>,
      );
      expect(wrapper.find(CheckmarkIcon)).toHaveLength(1);
    });
    it('Should return bell icon if status is deliver with alert', () => {
      const wrapper = shallow(
        <div>
          {stateToIcon(TRAFFIC_LIGHT_STATES.DELIVER_WITH_ALERT, 24, false)}
        </div>,
      );
      expect(wrapper.find(BellIcon)).toHaveLength(1);
    });
    it('Should return x icon if status is do not deliver', () => {
      const wrapper = shallow(
        <div>
          {stateToIcon(TRAFFIC_LIGHT_STATES.DO_NOT_DELIVER, 24, false)}
        </div>,
      );
      expect(wrapper.find(XIcon)).toHaveLength(1);
    });
    it('Should return white line icon if status is disabled', () => {
      const wrapper = shallow(
        <div>{stateToIcon(TRAFFIC_LIGHT_STATES.DISABLED, 24)}</div>,
      );
      expect(wrapper.find(WhiteLineIcon)).toHaveLength(1);
    });
    it('Should return white line icon if status is disabled and hide icon set to true', () => {
      const wrapper = shallow(
        <div>{stateToIcon(TRAFFIC_LIGHT_STATES.DISABLED, 24, true)}</div>,
      );
      expect(wrapper.find(WhiteLineIcon)).toHaveLength(0);
    });
  });
});
