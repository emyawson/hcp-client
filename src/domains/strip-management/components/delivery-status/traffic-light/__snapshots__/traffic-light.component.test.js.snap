// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Traffic light component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    status="DELIVER"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "border": false,
      "children": <styled.span>
        <Unknown
          height={24}
        />
      </styled.span>,
      "size": 60,
      "status": "DELIVER",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown
          height={24}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "height": 24,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "border": false,
        "children": <styled.span>
          <Unknown
            height={24}
          />
        </styled.span>,
        "size": 60,
        "status": "DELIVER",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            height={24}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "height": 24,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
